### DISCUSSION TITLE
Komponentspråk

### DISCUSSION THREAD

**Discussion by onordbo** [fremtind/jokul#2550] [Spørsmål] [answered]
## 🤔 Utfordring
I dag er Jøkul implementert på engelsk og dokumentert med engelske benevnelser, og norske beskrivelser i portalen. I biblioteket er det derimot en annen sak, der har vi en blanding av norske og engelske benevnelser. I mitt hode skaper dette potensiale for forvirring og scannbarhet når man leter etter komponenter man skal bruke.

Jeg er en stor forkjemper for at vi bruker norsk i fagspråket vårt, særlig når vi diskuterer med hverandre—og det burde vi absolutt fortsette med der vi kan og ønsker. Når det er sagt så skaper dagens løsning en kognitiv dissonans ved bruk av biblioteket, siden komponentbenevnelsen ikke matcher opp med hva som er teknisk implementert. Det er ingen etablert konvensjon _(såvidt meg kjent)_ som tilsier at man koder på noe annet språk enn engelsk, og jeg ser derfor ikke noen grunn eller ulempe ved å speile denne konvensjonen inne i biblioteket.

### Justere alle komponentbenevnelser i Figma til å matche den tekniske implementasjonen (engelsk).
Denne endringen vil muligens skape små ringvirkninger for brukerne av biblioteket på kort sikt _(typisk ved innsetting og tekstlig søk)_. Dog vil jeg si at det er en nytteverdig og langsiktig endring som vil skape en ryddigere brukeropplevelse, og at muskelminnet venner seg til nye benevnelser forholdsvis raskt.

⚠️ _**Obs:** Hvis ingen har noen sterke meninger om dette i løpet av uken (fra innleggsdato), vil endringen bli implementert snarest._

**Comment by Mikaila94** [comment]
Har du et eksempel på en benevnelse? Har ikke satt meg ned i dette problemet, men sånn rent teknisk implementasjon (med utvikler-perspektiv) så er det visse ting  jeg mener bør være engelsk. Også er det visse ting domene-spesifikke implementasjoner der jeg synes det kan være greit å bruke norsk. F.eks så ville jeg foretrukket "Button" over "Knapp". Og for domene-spesifikk foretrekker jeg for eksempel "OppsigelseApp" over "TerminationApp".

**Reply by onordbo** [reply]
Legger ved noen eksempler i screenshot.

I mitt hode så synes jeg benevnelse på faktiske komponenter burde være speilet på tvers av systemer, dog trenger man ikke _(i mitt hode)_ ha komponenter som heter "Knapp" i Figma for å kunne forholde seg til "knapper" som fagspråk generelt i dagligtalen. Når det er sagt så handler det mer om å være konsekvent, enn at det "må" være på engelsk, det kunne vært på norsk _(eller noe annet forsåvidt)_ så lenge det er likt der det "burde" være det, tenker jeg.

Korteste vei til produkt her på en nogenlunde logisk måte _(i mitt hode iaf)_ blir å stille benevnelser til engelsk, som vi har i systemet p.t _(i tillegg med hensyn til at visse ting i utviklerverdenen **burde** være på engelsk)_

![image](https://user-images.githubusercontent.com/4263132/143882140-40c53d84-69b7-43a3-94e9-7ca070a3cd2f.png)

**Reply by Mikaila94** [reply]
Skjønner,  da er jeg enig med korteste vei

**Comment by Murstam** [comment]
Jeg tenker det viktigste er at de har like navn i Figma og Jøkul, så da er det bare å bli enige om det skal være norsk eller engelsk. For min del kan det gjerne være norsk, men hvis dette gjør det vrient for utvikling tenker jeg at det er et godt nok argument for å kjøre engelsk.

**Comment by Slaine2** [comment, marked as answer]
For meg ville det også vært mer logisk om vi beholdt navngivningen lik overalt som mulig, og siden utviklingsspråket er engelsk, ville jeg holdt det likt i engelsk.

**Comment by Slaine2** [comment]
For å være konsekvent bør vi også navngi egenskapene i portalen på engelsk.

<img width="757" alt="Skjermbilde 2021-12-01 kl  14 40 24" src="https://user-images.githubusercontent.com/64286523/144244744-c51f3769-e51e-4a4c-9208-820f242e968d.png">
